### ISSUE TITLE
PHP Fatal error: LocalLockFactory

### ISSUE THREAD

**Issue by SirFlip** [shlinkio/shlink#585]
#### How Shlink is set-up

* Shlink Version: 1.20.2
* PHP Version: 7.4 (7.2 7.3)
* How do you serve Shlink: Self-hosted Apache
* Database engine used: MariaDB (10.4.11)

#### Summary

when using the cli I get a PHP fatal error. Discovered, because the "visit:locate" didn't work. The web api and the link resolving works fine as far as I know.

#### Current behavior

> **bin/cli visit:locate**
> PHP Fatal error:  Uncaught Error: Class 'Shlinkio\Shlink\LocalLockFactory' not found in /var/www/shlink/vendor/zendframework/zend-servicemanager/src/AbstractFactory/ConfigAbstractFactory.php:70
> Stack trace:
> #0 /var/www/shlink/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(764): Zend\ServiceManager\AbstractFactory\ConfigAbstractFactory->__invoke()
> #1 /var/www/shlink/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(200): Zend\ServiceManager\ServiceManager->doCreate()
> #2 [internal function]: Zend\ServiceManager\ServiceManager->get()
> #3 /var/www/shlink/vendor/zendframework/zend-servicemanager/src/AbstractFactory/ConfigAbstractFactory.php(68): array_map()
> #4 /var/www/shlink/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(764): Zend\ServiceManager\AbstractFactory\ConfigAbstractFactory->__invoke()
> #5 /var/www/shlink/vendor/zendframework/zend-servicemanager/src/ServiceManager.php(200): Zend\ServiceManager\ServiceManager->doCreate()
> #6 [internal function]: Zend\ServiceManager\ServiceManag in /var/www/shlink/vendor/zendframework/zend-servicemanager/src/AbstractFactory/ConfigAbstractFactory.php on line 70

#### How to reproduce

I don't know on other machines. For me its presistent on every command I tried.

**Comment by acelaya** [comment]
Thanks for reporting!

I will investigate and try to get a fix as soon as possible.

**Comment by acelaya** [comment]
@SirFlip I'm having some problems reproducing this bug.

I have tried a few combinations, but I see you reported the PHP version as `PHP Version: 7.4 (7.2 7.3)`.

What's exactly the version you used? Or did you try the three of them?

**Comment by SirFlip** [comment]
The main one is 7.4 but because of other stuff on the server I have the other ones as well and tried them out.

**Comment by acelaya** [comment]
I was asking because shlink is not yet compatible with PHP 7.4, and that's probably the reason it fails with that version.

However, it should work with the other two versions.

I will try to reproduce it with 7.4, just to make sure.

**Comment by acelaya** [comment]
Ok, I just reproduced it with all three versions. It seems the problem is on the dist file somehow, since it works with the official docker image :thinking: 

I will try to get it fixed.

**Comment by acelaya** [comment]
@SirFlip I found the root cause for this issue.

While a fix is provided, you can workaround it by doing this:

Open the `config/container.php` file and add this line:

```diff
 }
 
 // Build container
+class_alias(\Symfony\Component\Lock\Factory::class, 'Shlinkio\Shlink\LocalLockFactory');
 $config = require __DIR__ . '/config.php';
 $container = new ServiceManager($config['dependencies']);
 $container->setService('config', $config);
```

After that, the error should be gone.

**Comment by SirFlip** [comment]
Yes that works thank you very much for all your work.

Merry Christmas and a happy new year

**Comment by acelaya** [comment]
I have just released Shlink 1.20.3, hich includes this fix https://github.com/shlinkio/shlink/releases/tag/v1.20.3
